inst/shiny/tabs/nca.R

```diff
@@ -434,70 +454,65 @@ observeEvent(input$nca, {
+
+  # Filter only the analytes and doses requested
+  mydata$intervals <- mydata$intervals %>%
+    dplyr::filter(DOSNO %in% input$select_dosno,
+                  ANALYTE %in% input$select_analyte,
+                  PCSPEC %in% input$select_pcspec)
 
   # Define start imputations on intervals if specified by the user
   if (input$should_impute_c0) {
     mydata <- create_start_impute(mydata = mydata)
     mydata$impute <- "impute"
-
-  } else {
-    # Otherwise, the original intervals should start at C1 for all calculations
-    mydata <- PKNCA::PKNCAdata(
-      data.conc = mydata$conc,
-      data.dose = mydata$dose,
-      intervals = bind_rows(
-        format_pkncadata_intervals(
-          mydata$dose,
-          start_from_last_dose = FALSE
-        ),
-        intervals_userinput()
-      ) %>%
-        # Filter only the analytes and doses requested by the user
-        dplyr::filter(
-          DOSNO %in% input$select_dosno,
-          if ("EVID" %in% names(data())) EVID == 0 else TRUE
-        ),
-      units = PKNCA::pknca_units_table(
-        concu = mydata$conc$data$PCSTRESU[1],
-        doseu = mydata$dose$data$DOSEU[1],
-        amountu = mydata$conc$data$PCSTRESU[1],
-        timeu = mydata$conc$data$RRLTU[1]
-      )
-    )
   }
-  mydata(mydata)
 
-  # Perform NCA on the profiles selected
-  pk_nca_trigger(pk_nca_trigger() + 1)
-
-  # Update panel to show results page
-  updateTabsetPanel(session, "ncapanel", selected = "Results")
+  mydata(mydata)
 })
 
+# NCA running and obtention of results
+myres <- reactiveVal(NULL)
 res_nca <- reactiveVal(NULL)
-observeEvent(pk_nca_trigger(), {
+observeEvent(input$nca, {
   req(mydata())
-  withProgress(message = "Calculating NCA...", value = 0, {
-    myres <- PKNCA::pk.nca(data = mydata(), verbose = FALSE)
-
-    # Increment progress to 100% after NCA calculations are complete
-    incProgress(0.5, detail = "NCA calculations complete!")
 
-    # Make the starts and ends of results relative to last dose using the dose data
-    myres$result <- myres$result %>%
-      inner_join(select(mydata()$dose$data, -exclude)) %>%
-      mutate(start = start - !!sym(mydata()$dose$columns$time),
-             end = end - !!sym(mydata()$dose$columns$time)) %>%
-      select(names(myres$result))
-
-    # Return the result
-    res_nca(myres)
+  withProgress(message = "Calculating NCA...", value = 0, {
+    tryCatch({
+      myres <- PKNCA::pk.nca(data = mydata(), verbose = FALSE)
+
+      # Increment progress to 100% after NCA calculations are complete
+      incProgress(0.5, detail = "NCA calculations complete!")
+
+      # Make the starts and ends of results relative to last dose using the dose data
+      myres$result <- myres$result %>%
+        inner_join(select(mydata()$dose$data, -exclude)) %>%
+        dplyr::mutate(start = start - !!sym(mydata()$dose$columns$time),
+                      end = end - !!sym(mydata()$dose$columns$time)) %>%
+        dplyr::select(names(myres$result))
```

Comment:
nitpick: namespacing

```suggestion
        mutate(start = start - !!sym(mydata()$dose$columns$time),
                      end = end - !!sym(mydata()$dose$columns$time)) %>%
        select(names(myres$result))
```